lib/resolver.js

```diff
@@ -109,13 +109,43 @@ function SchemaResolver(rootSchema, external) {  // jshint ignore: line
     this.rootSchema = rootSchema;
     this.cache = {};
     this.resolved = null;
+    this.missing = [];
 
     this.resolvers = external && typeof external === 'object' ?
         getResolvers(external) :
         null;
+
```

Comment:
Why do you merge the `missing` arrays here? Elements to this are added only after calling `resolveRef()` and that does not happen before the constructor returns. This means you don't have anything to merge yet at this point?!